zendesk/macro.go

```diff
@@ -0,0 +1,165 @@
+package zendesk
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// Macro is information about zendesk macro
+type Macro struct {
+	Actions     []MacroAction `json:"actions"`
+	Active      bool          `json:"active"`
+	CreatedAt   time.Time     `json:"created_at"`
+	Description interface{}   `json:"description"`
+	ID          int64         `json:"id"`
+	Position    int           `json:"position"`
+	Restriction interface{}   `json:"restriction"`
+	Title       string        `json:"title"`
+	UpdatedAt   time.Time     `json:"updated_at"`
```

Comment:
It says [here](https://developer.zendesk.com/rest_api/docs/support/macros#json-format) that timestamps are `read-only=false`, but i think it weird 🤔 
I wonder if you could take a look for the response of write (create/update) operation about these fields?

Reply:
hi nukosuke, thanks for your review.

I tried `CreateMacro()` and `UpdateMacro()` using post.json (listed below).
There were no error but some properties, including timestamps, were ignored.

It may be justice that there are no errors, but this is certainly strange 🤔
Should the ignore field be set `omitempty` flag here?

----
post.json
```json
{
    "actions": [
        {
            "field": "status",
            "value": "solved"
        },
        {
            "field": "priority",
            "value": "normal"
        },
        {
            "field": "type",
            "value": "incident"
        },
        {
            "field": "assignee_id",
            "value": "current_user"
        },
        {
            "field": "group_id",
            "value": "current_groups"
        },
        {
            "field": "comment_value",
            "value": "Thanks for your request. This issue you reported is a known issue. For more information, please visit our forums. "
        }
    ],
    "url": "https://subdomain.zendesk.com/api/v2/macros/360111062754.json",
    "id": 360111062754,
    "title": "Close and redirect to topics",
    "active": true,
    "created_at": "2019-09-16T02:17:38Z",
    "updated_at": "2022-09-16T02:17:38Z",
    "position": 9999,
    "description": null,
    "restriction": null
}
```

`diff post.json created.json` are:
```diff
28,29c28,29
<     "url": "https://subdomain.zendesk.com/api/v2/macros/360111062754.json",
<     "id": 360111062754,
---
>     "url": "https://MYZENDESK.zendesk.com/api/v2/macros/360191461852.json",
>     "id": 360191461852,
32,34c32,34
<     "created_at": "2019-09-16T02:17:38Z",
<     "updated_at": "2022-09-16T02:17:38Z",
<     "position": 9999,
---
>     "updated_at": "2021-01-28T00:14:53Z",
>     "created_at": "2021-01-28T00:14:53Z",
>     "position": 10038,
```

Reply:
Thank you for that 🙏 

yeah, that behavior makes sense. They seems to be ignored by API's validator like StrongParameters. I think we could omit keys of that kind of fields because it's just a doc problem.

Reply:
Fix it.
thank you for your guide.